sw/device/lib/testing/sysrst_ctrl_testutils.c

```diff
@@ -39,30 +40,49 @@ void setup_dio_pins(dif_pinmux_t *pinmux, dif_sysrst_ctrl_t *sysrst_ctrl) {
   CHECK_DIF_OK(dif_pinmux_pad_write_attrs(
       pinmux, kTopEarlgreyDirectPadsSysrstCtrlAonFlashWpL, kDifPinmuxPadKindDio,
       in_attr, &out_attr));
+}
+
+void sysrst_ctrl_testutils_release_dio(dif_sysrst_ctrl_t *sysrst_ctrl,
+                                       bool release_ec, bool release_flash) {
+  // Make sure the the DIO pins EC reset and flash WP are released according to
+  // the documentation:
+  // https://opentitan.org/book/hw/ip/sysrst_ctrl/doc/theory_of_operation.html#ec-and-power-on-reset
+  // https://opentitan.org/book/hw/ip/sysrst_ctrl/doc/theory_of_operation.html#flash-write-protect-output
   // We also need to disable the output override mecanism (i.e. "release the
   // pin").
-  dif_sysrst_ctrl_pin_config_t cfg_ec_reset = {
-      .enabled = kDifToggleDisabled,
-      .allow_zero = true,
-      .allow_one = true,
-      .override_value = false,
-  };
-  CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
-      sysrst_ctrl, kDifSysrstCtrlPinEcResetInOut, cfg_ec_reset));
+  if (release_ec) {
+    dif_sysrst_ctrl_pin_config_t cfg_ec_reset = {
+        .enabled = kDifToggleDisabled,
+        .allow_zero = true,
+        .allow_one = true,
+        .override_value = false,
+    };
+    CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
+        sysrst_ctrl, kDifSysrstCtrlPinEcResetInOut, cfg_ec_reset));
+  }
   // Confusingly, the flash WP is different: the value of flash_wp_l_o defaults
   // to logic 0 when it is not explicitly driven via the override function.
   // Therefore to disable the driver we need to *enable* the override and set
   // to 1.
-  dif_sysrst_ctrl_pin_config_t cfg_flash_wp = {
-      .enabled = kDifToggleEnabled,
-      .allow_zero = false,
-      .allow_one = true,
-      .override_value = true,
-  };
-  CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
-      sysrst_ctrl, kDifSysrstCtrlPinFlashWriteProtectInOut, cfg_flash_wp));
-  // Disable the EC reset debounce timer because it could interfere with
-  // reading the pins.
+  if (release_flash) {
+    dif_sysrst_ctrl_pin_config_t cfg_flash_wp = {
+        .enabled = kDifToggleEnabled,
+        .allow_zero = false,
+        .allow_one = true,
+        .override_value = true,
+    };
+    CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
+        sysrst_ctrl, kDifSysrstCtrlPinFlashWriteProtectInOut, cfg_flash_wp));
+  }
+}
+
+void sysrst_ctrl_testutils_set_ec_rst_pulse_width(
+    dif_sysrst_ctrl_t *sysrst_ctrl, uint32_t pulse_us) {
+  // The register field counts in aon_clock ticks.
+  uint64_t ticks =
+      udiv64_slow((uint64_t)pulse_us * kClockFreqAonHz, 1000 * 1000, NULL);
+  // The register field is 16-bit wide.
+  CHECK(ticks <= 0xffff);
```

Comment:
UINT16_MAX?

Reply:
I don't think this would be suitable since it's a field mask. I have replaced it with the mask from the autogenerated headers.